### DISCUSSION TITLE
想参与 3.0 版本的开发吗？来报名吧！😎

### DISCUSSION THREAD

**Discussion by kuitos** [umijs/qiankun#2688] [Announcements]
大家好，近期我们发布了 qiankun 3.0 的 rc 版本，你可以在这里找到发布信息：[qiankun 3.0.0-rc.0](https://github.com/umijs/qiankun/releases/tag/v3.0.0-rc.0) 🎉

但 rc 版本还有很多待完善的功能，我们非常期待能与社区的伙伴们一起共同构建 qiankun 3.0 的生态，与社区一起发布最终的正式版。为此，我们整理并发布了一些与 3.0 相关的任务。我们希望能够吸引更多对此感兴趣的同学们加入到 3.0 版本的开发中来🥳。

如果你对参与这些任务感兴趣，可以在任务列表中查看具体内容：[qiankun 3.0 Milestone](https://github.com/umijs/qiankun/milestone/3)

### 领取方式
领取任务的方式很简单，只需要在本帖下**回复你想领取的任务链接，并提供你能够开始任务的最早时间**，我们会按照报名的顺序及时间安排来决定参与的人选（qiankun contributor 优先），并将其添加到 qiankun 开发者群组中。

### 参与要求
1. 请确保你有足够的开发时间。我们期望任务负责人能在一周内开始，在一个月内完成领取功能 80% 以上的开发工作。
2. 请确保你能够进行基本的后续维护工作。也就是说，我们期望你能在接下来的半年内负责你所开发功能的 bug 修复和社区维护工作。

### 你将获得
1. 参与到社区最流行的微前端解决方案之一的全新版本建设中🤳
2. 成为 qiankun team member，免费获取 github copilot、jetbrains ide 全家桶等工具的开源版本 licence🤠

### 任务工作流
> **3.0 在 next 分支研发**
1. 任务领取后，task assign 给相关同学
2. task 内沟通设计思路 & 方案
4. 提交 pr
5. 评审通过、合并 pr

希望能够吸引到更多的开发者加入到 qiankun 3.0 版本的开发中来，共同推动项目的发展！期待你的加入~

**Comment by bravepg** [comment]
本周可以开始 https://github.com/umijs/qiankun/issues/2677

**Comment by hakushinring** [comment]
[报名cli开发任务](https://github.com/umijs/qiankun/issues/2679)，有cli开发经验，通读过qiankun2.0代码，最早开始时间10.02，可长期维护

**Reply by kuitos** [reply]
Nice~ 这个能早点开始么，可以先在 issue 下聊下方案跟设计，代码可以后面再写，10.2 开始有点晚了😂

**Reply by hakushinring** [reply]
方案和设计可以23号开始

**Reply by hakushinring** [reply]
我先明确下我要产出的内容 ，代码开发前产出  
功能清单（包括交互逻辑，参数，模板内容），技术实现方案（设计文档）  
这样ok吗

**Reply by kuitos** [reply]
OK 的

**Reply by qiYuei** [reply]
目的：脚手架是希望提供工具给用户通过一系列选择就可以生成 主子应用怎么接入乾坤的示例代码。
功能：
   1. 支持单独创建主应用/子应用,亦支持同时创建主应用和子应用
   2. 预置 `qiankun` 必要配置信息
   3. 主子应用同时创建时自动注入子应用信息至主应用

实现：
   - 继承各大 `cli` 工具（vue-cli ,create-vue,create-react-app,create-umi）

**Reply by trueLoving** [reply]
这个任务还缺人吗 有 CLI 开发经验 想参与一起开发

**Comment by nayonglin** [comment]
报名[#2678 ](https://github.com/umijs/qiankun/issues/2678) ，有webpack插件开发经验，本周可以开始，可长期维护，方案和细节怎么沟通呀？

**Reply by kuitos** [reply]
👍直接在issue里沟通就行

**Reply by nayonglin** [reply]
好的

**Comment by kiner-tang** [comment]
想要报名，现在还有哪些任务没有被认领的

**Reply by kiner-tang** [reply]
#2684 这个有同学认领不，没有的话我领一下

**Reply by kuitos** [reply]
OK，可以 issue 里聊下方案

**Reply by kiner-tang** [reply]
好的，我这几天先捋捋，之后把方案同步到 issue

**Comment by gjmhandsome** [comment]
想要报名#2686，下周开始，有人领了吗

**Reply by kuitos** [reply]
需要能长期维护（半年起），这个 OK 的不？

**Reply by gjmhandsome** [reply]
ok

**Comment by qiongshusheng** [comment]
认领一下 #2683

**Reply by ZKunZhang** [reply]
我可以跟你一起吗？我也想参与 qiankun 的一些事情

**Comment by ZKunZhang** [comment]
认领一下 #2687

**Reply by kuitos** [reply]
欢迎👏 重点看一下参与要求哈，需要保证至少半年能持续维护

**Reply by ZKunZhang** [reply]
我会的，公司里在持续使用 qiankun 并对 qiankun3.0 有很高的期待，我也希望能多参与开发解决公司内的痛点的同时，给这个框架提些自己的建议。

**Reply by ZKunZhang** [reply]
抱歉，对不起，一直没时间参与了。我的就业状态有些不稳定了。导致一直在工作的事情上焦头烂额的。

**Comment by linghaoSu** [comment]
认领 https://github.com/umijs/qiankun/issues/2689 这个 vue 的任务

**Reply by kuitos** [reply]
需要确保至少半年的长期维护，这个OK的么？

**Reply by linghaoSu** [reply]
维护时间应该没有问题的

**Comment by snvtac** [comment]
认领 #2684 这个任务，目前可一周内开始，可长期维护。

**Comment by wanghangit** [comment]
想认领[#2604 ]这个任务，维护时间没有问题

**Comment by ponyfly** [comment]
想认领 #2806 目前可一周内开始，可长期维护

**Comment by wanghangit** [comment]
想认领https://github.com/umijs/qiankun/issues/2694 这个任务，可长期维护

**Comment by GGwujun** [comment]
想认领https://github.com/umijs/qiankun/issues/2759 和 https://github.com/umijs/qiankun/issues/2823 任务，可长期维护

**Reply by kuitos** [reply]
#2759 已经有 pr 了，#2823 你可以直接在 issue 下沟通，可以先发下你的 RFC

**Comment by dleged** [comment]
领取对增强 fetch 的 task: https://github.com/umijs/qiankun/issues/2914

**Reply by kuitos** [reply]
这个任务我已经在做了，上周忘记更新状态了😂 你可以看下其他任务？

**Reply by dleged** [reply]
好的哈

**Reply by dleged** [reply]
https://github.com/umijs/qiankun/issues/2681 E2E 这个呢

**Reply by kuitos** [reply]
这个还没人领
